src/Sylius/Component/Channel/Context/EnvBased/ChannelContext.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Channel\Context\EnvBased;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Context\ChannelNotFoundException;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+
+final class ChannelContext implements ChannelContextInterface
+{
+    public function __construct(
+        private ChannelRepositoryInterface $channelRepository,
+        private ?string $channelCodeFromEnv,
+    ) {
+    }
+
+    public function getChannel(): ChannelInterface
+    {
+        if ('cli' !== PHP_SAPI || null === $this->channelCodeFromEnv) {
```

Comment:
Since it'll work virtually always, is there any point in limiting it only to cli?

Reply:
Yes, otherwise you risk it being used in http context. The env var might be defined globally either by accident or on purpose.

Reply:
@vvasiloi is right, it was made intentionally to prevent mistakes, also I don't see any use-case outside the CLI context.

Reply:
I see the potential for it - if you would like to route your websites to given channels based on your server configuration rather then host name resolution in the store